ironic/drivers/modules/drac/inspect.py

```diff
@@ -35,16 +41,164 @@
 
 METRICS = metrics_utils.get_metrics_logger(__name__)
 
+sushy = importutils.try_import('sushy')
+
+# System Configuration Tag Constant
+_SYSTEM_CONFIG_TAG = '<SystemConfiguration Model'
+
+# Time Out Constant
+_TIMEOUT_SECONDS = 120
+
+# Job Id Url Constant
+_JOB_ID_URL = '/redfish/v1/Managers/iDRAC.Embedded.1/Actions/Oem/' \
+              'EID_674_Manager.ExportSystemConfiguration'
+
+# Xml Data Url Constant
+_XML_DATA_URL = '/redfish/v1/TaskService/Tasks/'
+
+# Mac Address Url Constant
+_MAC_URL = '/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/'
+
+# Payload Constant
+_PAYLOAD = {'ExportFormat': 'XML', 'ShareParameters': {'Target': 'NIC'}}
+
+# Headers Constant
+_HEADERS = {'content-type': 'application/json'}
+
+# Job Response Code Constant
+_JOB_RESPONSE_CODE = 202
+
 
 class DracRedfishInspect(redfish_inspect.RedfishInspect):
-    """iDRAC Redfish interface for inspection-related actions.
+    """iDRAC Redfish interface for inspection-related actions."""
 
-    Presently, this class entirely defers to its base class, a generic,
-    vendor-independent Redfish interface. Future resolution of Dell EMC-
-    specific incompatibilities and introduction of vendor value added
-    should be implemented by this class.
-    """
-    pass
+    def inspect_hardware(self, task):
+        """Inspect hardware to get the hardware properties.
+
+        Inspects hardware to get the essential properties.
+        It fails if any of the essential properties
+        are not received from the node.
+
+        :param task: a TaskManager instance.
+        :raises: HardwareInspectionFailure if essential properties
+                 could not be retrieved successfully.
+        :returns: The resulting state of inspection.
+
+        """
+        super(DracRedfishInspect, self).inspect_hardware(task)
+        node = task.node
+        system = redfish_utils.get_system(node)
+        pxe_dev_nics = self._get_pxe_dev_nics(task, system)
+        if pxe_dev_nics is None:
+            LOG.warning("No PXE enabled NIC was found for node "
+                        "%(node_uuid)s.", {'node_uuid': node.uuid})
+
+        if (system.ethernet_interfaces
+                and system.ethernet_interfaces.summary):
+            macs = system.ethernet_interfaces.summary
+
+            enabled_macs = {nic_mac: nic_state
+                            for nic_mac, nic_state in macs.items()
+                            if nic_state == sushy.STATE_ENABLED}
+            if enabled_macs:
```

Comment:
I think we could make this even simpler by removing the highlighted code and instead getting the ports for the node using:
objects.Port.list_by_node_id(task.context, node.id)
We could then iterate through those ports below and check to see if the mac address on the port is in pxe_dev_nics

Reply:
Done